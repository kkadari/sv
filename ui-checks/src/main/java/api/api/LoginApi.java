package api.api;

import io.restassured.response.Response;

import static io.restassured.RestAssured.given;

public class LoginApi {

    public static Response login(String baseUrl, String username, String password){
        return given()
                .params("username", username, "password", password)
               .when()
                .post(baseUrl + "/cs_login");
    }
}
