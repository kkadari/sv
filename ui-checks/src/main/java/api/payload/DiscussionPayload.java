package api.payload;

public class DiscussionPayload {

//    {
//        "subject": "asdasdasads",
//        "markAsQuestion": false,
//        "body": "<body><p>asdasdasdasdasasdas</p></body>",
//        "handlingLevel": "2",
//        "publishBar": {
//            "visibility": "all"
//        },
//        "notAttributable": false,
//    }

    private String subject;
    private Boolean markAsQuestion;
    private String body;
    private String handlingLevel;
    private PublishPayload publishBar;
    private boolean notAttributable;

    public DiscussionPayload(String subject, Boolean markAsQuestion, String body, String handlingLevel, boolean notAttributable) {
        this.subject = subject;
        this.markAsQuestion = markAsQuestion;
        this.body = body;
        this.handlingLevel = handlingLevel;
        this.publishBar = new PublishPayload();
        this.notAttributable = notAttributable;
    }

    public String getSubject() {
        return subject;
    }

    public Boolean getMarkAsQuestion() {
        return markAsQuestion;
    }

    public String getBody() {
        return body;
    }

    public String getHandlingLevel() {
        return handlingLevel;
    }

    public PublishPayload getPublishBar() {
        return publishBar;
    }

    public boolean isNotAttributable() {
        return notAttributable;
    }
}
