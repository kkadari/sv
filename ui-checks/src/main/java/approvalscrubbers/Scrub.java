package approvalscrubbers;

import java.util.ArrayList;

public class Scrub {

    public static String scrub(ArrayList<String> scrubbers, String content){
        for(String scrubber : scrubbers){
            content = content.replaceAll(scrubber, "***scrubbed***");
        }

        return content;
    }

}
