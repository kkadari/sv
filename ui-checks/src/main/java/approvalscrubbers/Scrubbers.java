package approvalscrubbers;

public class Scrubbers {

    public static String scrubJiveId(){
        return "id=\"jive-[0-9]*\"";
    }

    public static String scrubThreadRef(){
        return "\\/thread\\/[0-9]*";
    }

    public static String scrubDisplayObjectKeys(){
        return "name=\"displayedObjectKeys\" value=\"[0-9_,]*\"";
    }

}
