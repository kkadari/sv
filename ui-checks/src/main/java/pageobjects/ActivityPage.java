package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

import java.util.List;

public class ActivityPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-act-title")
    List<WebElement> divTitles;

    public ActivityPage(WebDriver driver){
        super(driver);
    }

    public String returnStreamListHtml(){
        String streams = "";

        for(WebElement title : divTitles){
            streams += title.getAttribute("outerHTML");
        }

        return streams;
    }

}
