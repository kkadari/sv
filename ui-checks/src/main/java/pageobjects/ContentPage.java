package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.List;

public class ContentPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-td-title")
    List<WebElement> tableCellTitle;

    public ContentPage(WebDriver driver){
        super(driver);
    }

    public String returnTableRowContentHtml(){
        WebDriverWait wait = new WebDriverWait(driver, 10);
        wait.until(ExpectedConditions.visibilityOf(tableCellTitle.get(0)));

        String tableRowToReturn = "";

        for(WebElement row : tableCellTitle){
            tableRowToReturn += row.getAttribute("outerHTML");
        }

        return tableRowToReturn;
    }

}
