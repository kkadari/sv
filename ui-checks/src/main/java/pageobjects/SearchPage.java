package pageobjects;

import org.openqa.selenium.NoSuchElementException;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

public class SearchPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-content-results")
    WebElement divMainResults;

    @FindBy(how = How.CSS, using = ".j-search-result-summary")
    WebElement divResultContent;

    public SearchPage(WebDriver driver){
        super(driver);
    }

    public String returnSearchResults(){
        return divMainResults.getAttribute("outerHTML");
    }

    public boolean returnResultIsDisplayed(){
        try {
            return divResultContent.isDisplayed();
        } catch (NoSuchElementException e){
            return false;
        }
    }

}
