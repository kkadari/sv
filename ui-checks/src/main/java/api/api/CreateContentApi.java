package api.api;

import api.payload.DiscussionPayload;
import io.restassured.http.ContentType;
import io.restassured.http.Cookies;

import static io.restassured.RestAssured.given;

public class CreateContentApi {

    public static void createDiscussion(Cookies cookies, String baseUrl, DiscussionPayload discussionPayload){
        given()
            .contentType(ContentType.JSON)
            .cookies(cookies)
            .body(discussionPayload)
        .when()
            .post(baseUrl + "/__services/v2/rest/discussion");
    }
}
