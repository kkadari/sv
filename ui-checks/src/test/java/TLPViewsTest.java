import api.api.CreateContentApi;
import api.payload.DiscussionPayload;
import approvalscrubbers.Scrub;
import approvalscrubbers.Scrubbers;
import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Test;
import pageobjects.ActivityPage;
import pageobjects.ContentPage;
import pageobjects.SearchPage;

import java.util.ArrayList;

public class TLPViewsTest extends TestSetup {

    @Before
    public void generateNineEntries(){
        for(int i = 0; i < 9; i++){
            int tlp = (i % 4) + 1;
            DiscussionPayload discussionPayload = new DiscussionPayload("Check discussion", false, "Generated for automation", Integer.toString(tlp), false);

            CreateContentApi.createDiscussion(cookies, baseUrl, discussionPayload);
        }
    }

    @Test
    public void approveTlpPillsInAdvancedSearch() throws InterruptedException {
        SearchPage search = new SearchPage(driver);

        Boolean resultsShown = false;
        while (!resultsShown){
            driver.navigate().to(baseUrl + "/search.jspa?q=Check discussion");

            resultsShown = search.returnResultIsDisplayed();
        }

        String rawAdvancedSearch = search.returnSearchResults();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
            add(Scrubbers.scrubSearchUrl());
            add(Scrubbers.scrubDates());
            add(Scrubbers.scrubMessageId());
        }};

        String scrubbedAdvancedSearch = Scrub.scrub(scrubbers, rawAdvancedSearch);

        Approvals.verifyHtml(scrubbedAdvancedSearch);
    }

    @Test
    public void approveTlpPillsInActivityFeed(){
        driver.navigate().to(baseUrl + "/news?channel=recent");

        ActivityPage activity = new ActivityPage(driver);
        String rawStreamList = activity.returnStreamListHtml();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
           add(Scrubbers.scrubDates());
           add(Scrubbers.scrubActTitleId());
           add(Scrubbers.scrubThreadId());
        }};

        String scrubbedStreamList = Scrub.scrub(scrubbers, rawStreamList);

        Approvals.verifyHtml(scrubbedStreamList);
    }

    @Test
    public void approveTlpPillsInContentFeed(){
        driver.navigate().to(baseUrl + "/content?filterID=all&itemView=detail");

        ContentPage content = new ContentPage(driver);
        String rawRowContentList = content.returnTableRowContentHtml();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
            add(Scrubbers.scrubDates());
            add(Scrubbers.scrubThreadId());
            add(Scrubbers.scrubDataObjectId());
        }};

        String scrubbedRowContentList = Scrub.scrub(scrubbers, rawRowContentList);

        Approvals.verifyHtml(scrubbedRowContentList);
    }

}
