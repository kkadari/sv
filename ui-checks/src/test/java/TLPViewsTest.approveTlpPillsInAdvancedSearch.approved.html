<ol class="j-search-results j-content-results"><li data-self="http://localhost:8080/api/core/v3/***scrubbed***" class="j-search-result j-content "><div class="j-result-icon"><a ***scrubbed***>Load more items</span></a></li></ol>
